api/src/routes/datasets.js

```diff
@@ -337,13 +337,61 @@ router.get(
 
 // todo - register routes specific to dataset uploads optionally based on verifyUploadEnabledForRole middleware
 function verifyUploadEnabledForRole(req, res, next) {
```

Comment:
Added more conditions for verifying if a feature is enabled on the API-side via feature flags